columnphysics/icepack_atmo.F90

```diff
@@ -916,14 +896,28 @@ subroutine icepack_atm_boundary(sfctype,                   &
       real (kind=dbl_kind) :: &
          l_uvel, l_vvel, l_Uref
 
-      real (kind=dbl_kind), dimension(:), allocatable :: &
-         l_Qa_iso, l_Qref_iso   ! local copies of Qa_iso, Qref_iso
-
-      logical (kind=log_kind) :: &
-         iso_flag  ! flag to turn on iso calcs in other subroutines
+      logical (kind=log_kind), save :: &
+         first_call_ice = .true.   ! first call flag
 
       character(len=*),parameter :: subname='(icepack_atm_boundary)'
 
+      !------------------------------------------------------------
+      ! Check optional arguments
+      ! This is used for ocean and ice sfctype, need to be careful
+      !------------------------------------------------------------
+
+      if (sfctype == 'ice') then
+      if (argcheck == 'always' .or. (argcheck == 'first' .and. first_call_ice)) then
```

Comment:
The logic in this line is duplicated a few times. Would it make sense to isolate it to a function somewhere ? Then we could `use` that function instead of the `argcheck` variable.

The function could receive that boolean `fist_call` (or here `first_call_ice`) as argument and return a boolean...